perch-agent/src/api/agent_routes.py

```
@@ -0,0 +1,200 @@
+# at the cap is 500k chars per request — enough to push the LLM into a
+# slow/expensive turn or DOS one pod's CPU/memory at the prompt-template
+# render. Reject early with 422 when the sum exceeds this.
+_TOTAL_CONTENT_LIMIT = 60_000
+
+
+class ChatRequest(BaseModel):
+    messages: list[ChatMessage] = Field(min_length=1, max_length=50)
+    scope: ChatScope | None = None
+
+    @model_validator(mode="after")
+    def _validate_total_content(self) -> "ChatRequest":
```

Ruff (0.15.12)

[warning] Remove quotes from type annotation

Remove quotes

(UP037)